agent/agent.go

```diff
@@ -302,28 +316,13 @@ func NewAgent(options ...Option) (*Agent, error) {
 		// Log the error in the current span
 		OnSpanFinishPanic: scopeError.LogErrorInRawSpan,
 	})
-
+	agent.SetTestingMode(agent.testingMode)
```

Comment:
For another PR: remove this and do it directly on `NewSpanRecorder`

Reply:
Ok